### ISSUE TITLE
Filtering by Port

### ISSUE THREAD

**Issue by HaydnTrigg** [danielkrainas/node-netstat#13]
I am trying to export only the processes running on port 3000. Using the code

```
netstat({
    filter: {
        protocol: 'tcp',
        local: {port:3000}
    },
    limit: 1
}, function (data) {
    console.log(data)
});
```

I am using Windows 10. Removing the `local: {port: 3000}` returns the following results:

```
{ protocol: 'tcp',
  local: { port: 135, address: null },
  remote: { port: 0, address: null },
  state: 'LISTENING',
  pid: 912 }
{ protocol: 'tcp',
  local: { port: 445, address: null },
  remote: { port: 0, address: null },
  state: 'LISTENING',
  pid: 4 }
{ protocol: 'tcp',
  local: { port: 623, address: null },
  remote: { port: 0, address: null },
  state: 'LISTENING',
  pid: 11628 }
{ protocol: 'tcp',
  local: { port: 1301, address: null },
  remote: { port: 0, address: null },
  state: 'LISTENING',
  pid: 9992 }
{ protocol: 'tcp',
  local: { port: 3000, address: null },
  remote: { port: 0, address: null },
  state: 'LISTENING',
  pid: 8556 }
```

**Comment by danielkrainas** [comment]
Thanks! Definitely a bug. The filtering code doesn't do a deep compare. I'll fix it

**Comment by danielkrainas** [comment]
Released in [v1.3.3](https://github.com/danielkrainas/node-netstat/releases/tag/v1.3.3)
